documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+
+Key areas covered include:
+
+* Authorization flows:
+
+    * Authorization Code Flow.
+    * Client Initiated Back Channel Authentication (CIBA).
+    * Client Credentials.
+
+* Scope Parameter. Definition of scope parameter format applicable to CAMARA APIs.
+
+* ID Token Content. CAMARA specific details about the content of the Id token.
+
+* Client Authentication. Specifications for client authentication within the CAMARA OIDC profile.
+
+By encapsulating these elements within the CAMARA OIDC profile, this document aims to provide a comprehensive guide for developers and operators, ensuring consistent implementation and adherence to standardized security measures across the CAMARA ecosystem. The defined OIDC profile not only facilitates the integration process, but also serves as a basic framework for developers wishing to leverage the CAMARA APIs while maintaining security and interoperability.
+
+
+## Audience
+
+The target audience for this document is the service/technical departments of operators exposing network functions via standard CAMARA APIs. And applications or client systems that consume CAMARA standard APIs to make use of the operator's network capabilities.
+
+
+## Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
+## Authorization using the Authorization Code Flow
+
+The [OAuth 2.0 Authorization Code](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1) flow involves exchanging an authorization code for a token. Instead of requesting authorization directly from the resource owner, the client directs the resource owner to an authorization server (via its user-agent), which in turn directs the resource owner back to the client with the authorization code.
+
+### Authentication Request
+
+An Authentication Request is an [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html) Authorization Request that requests that the End-User be authenticated by the Authorization Server.
+
+The CAMARA profile uses the following OpenID Connect request parameters with the Authorization Code Flow:
+
+* **scope**
+
+  REQUIRED. A space delimited and a case-sensitive list of ASCII strings for OAuth 2.0 scope values. OpenID Connect requests MUST contain the `openid` scope value. If the `openid` scope value is not present, the ID Token will not be returned. Other scope values MAY be present. Scope values used that are not understood by an implementation SHOULD be ignored. More information about scope values as described in [The Scope Parameter Section](#the-scope-parameter).
```

Comment:
Either the `openId` scope value is mandatory, or it is not.

If you are defining different behaviour when the `openid` scope value is not present, then that is an additional authorisation scheme whose baseline is the OAuth2 authorisation code flow, and not the OpenId Connect Core authorisation code flow.

So the authorisation code flow section should be separated into two parts:
- Authorisation code flow with `openId` scope value, with OpenId Connect Core as the baseline
- Authorisation code flow without `openId` scope value, with RFC 6749 as the baseline